File: solutions/typescript/darts/1/darts.ts
const INNER_RADIUS = 1;
const INNER_POINTS = 10;
const MID_RADIUS = 5;
const MID_POINTS = 5;
const OUTER_RADIUS = 10;
const OUTER_POINTS = 1;

export function score(x: number, y: number): number {
  const radius = getRadius(x, y);
  if (isInnerCircle(radius))
    return INNER_POINTS;
  if (isMidCircle(radius))
    return MID_POINTS;
  if (isOuterCircle(radius))
    return OUTER_POINTS;
  return 0;
}

function isInnerCircle(radius: number): boolean {
  return radius <= INNER_RADIUS;
}
function isMidCircle(radius: number): boolean {
  return radius <= MID_RADIUS;
}
function isOuterCircle(radius: number): boolean {
  return radius <= OUTER_RADIUS;
}

function getRadius(x: number, y: number): number {
  return Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2));
}

File: solutions/csharp/darts/1/Darts.cs
using System;
public static class Darts
{
    private static readonly int _innerCircleRadius = 1;
    private static readonly int _middleCircleRadius = 5;
    private static readonly int _outerCircleRadius = 10;

    private static readonly int _innerPoints = 10;
    private static readonly int _middlePoints = 5;
    private static readonly int _outerPoints = 1;
    private static readonly int _outsidePoints = 0;

    private static double GetRadius(double x, double y)
    {
        return Math.Sqrt(x * x + y * y);
    }

    private static bool IsInnerCircle(double radius)
    {
        return radius <= _innerCircleRadius;
    }

    private static bool IsMiddleCircle(double radius)
    {
        return radius <= _middleCircleRadius;
    }

    private static bool IsOuterCircle(double radius)
    {
        return radius <= _outerCircleRadius;
    }

    public static int Score(double x, double y)
    {
        double radius = GetRadius(x, y);
        if (IsInnerCircle(radius))
        {
            return _innerPoints;
        }
        if (IsMiddleCircle(radius))
        {
            return _middlePoints;
        }
        if (IsOuterCircle(radius))
        {
            return _outerPoints;
        }
        return _outsidePoints;
    }
}
